code/components/citizen-server-impl/src/state/ServerGameState.cpp

```diff
@@ -960,6 +960,23 @@ void ServerGameState::Tick(fx::ServerInstanceBase* instance)
 					}
 				}
 			}
+			
+			if (entity->type == sync::NetObjEntityType::Player)
+			{
+				auto entityHealth = entity->syncTree->GetPedHealth();
+				auto ownerRef = entity->GetClient();
+
+				if (entityHealth->health <= 0 && !entityHealth->isDead)
```

Comment:
Peds have a [fatal injury threshold](https://docs.fivem.net/natives/?_0xD839450756ED5A80) under which they are considered dead. This doesn't have to be 0.

Reply:
Yes, but this native is only client side and since when a player is dead the life is at 0 I decided to check the health. But are they any scenarios that the health is not 0 and your dead?

Reply:
Just tested this script on a blank txadmin server. Setting the player's health to anything below 100 kills them

```lua
Citizen.CreateThread(function()
  Wait(6000)
  SetEntityHealth(GetPlayerPed(-1), 99);
end)
```

Reply:
Nevermind, it seems that something sets your health to 0 as part of the dying process. In this image I set it to 99 and it went to 0. This is probably not an issue then.
![image](https://github.com/citizenfx/fivem/assets/1671056/ada7f86f-6d92-4f7c-9516-addb39e39971)

Reply:
Yes, but it will work anyway since if you die your health will be automatically set to 0.